### PR TITLE
improvement: setup contract model

### PR DESCRIPTION
**Pull request by dsinghvi**
An attempt to take fields listed [`here`](https://docs.google.com/document/d/1ZbxaaZhBaQGA0vgJCEH5skbcA0JFZAlWrh2VvTHsFdE/edit) and formalize into a smart contract.

### CODE REVIEW

File: contracts/Terms.sol [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ */
+abstract contract PaymentTerms {
+
+    function getPaymentAmount() public virtual view returns(uint);
+
+}
+
+contract FlatFee is PaymentTerms {
+
+    // TODO: How can we generalize this to other tokens and stable coins?
```
Comment (brijeshrakholia) [inline contracts/Terms.sol:36]: In Solidity, we can probably only use ETH as a transaction gas. We might be able to write up some contracts that allows us to use values that represent other cryptocurrencies. That's a good question..

File: contracts/Terms.sol [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+}
+
+// If all parties agree on task completion, then the transaction will be paid out.
+contract UnanimousVote is ConsensusType {
+
+    bool employerVote = false;
+
+    bool employeeVote = false;
+
```
Comment (brijeshrakholia) [inline contracts/Terms.sol:80]: Does `employerVote` and `employeeVote` remain accessible by other contracts? If not, do we need `getEmployeeVote()` and `getEmployerVote()`?

File: contracts/Terms.sol (hunk 1 above)
Comment (dsinghvi) [reply contracts/Terms.sol:36]: Got it -- someone else mentioned using `payment streams` and how that could allow you to send and receive erc20 tokens like stable coins. If we have enough time, could be cool to come back and see what's involved.

File: contracts/Terms.sol (hunk 2 above)
Comment (dsinghvi) [reply contracts/Terms.sol:80]: Why do the other contracts need to know those? they just need to know if they agreed right?

File: contracts/Terms.sol [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    string taskDescription;
+
+    // Unix timestamp for expiry date
+    uint256 expiry; 
+
+    // TODO: Support > 1 employers pooling their funds together
+    address employerAddr;
+
+    // TODO: Support > 1 contractors receiving payment
+    address contractorAddr;
```
Comment (akmahaja) [inline contracts/Terms.sol:16]: For pooling 

`address[] public employerAddr`

File: contracts/Terms.sol (hunk 2 above)
Comment (brijeshrakholia) [reply contracts/Terms.sol:80]: Yeah that's true. I can't think of any use case for them.

**Review by brijeshrakholia — APPROVED** [review]

**Pull request merged**